src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -731,6 +731,7 @@ protected bool TakeStartLine(SocketInput input)
                     // URI was encoded, unescape and then parse as utf8
                     pathEnd = UrlPathDecoder.Unescape(pathBegin, pathEnd);
                     requestUrlPath = pathBegin.GetUtf8String(pathEnd);
+                    requestUrlPath = PathNormalizer.NormalizeToNFC(requestUrlPath);
```

Comment:
It's a lot of work. Checking if the string needs normalization requires checking some large tables of Unicode characters. Also, I'd basically have to re-implement `string.Normalize()`, and that wouldn't even save us an allocation because the native code it calls requires an output buffer to put the results i.e. it's not done in place.